admin_app_2/src/components/ContentModal.tsx

```diff
@@ -179,8 +184,7 @@ const DeleteContentModal = ({
         <Button
           onClick={() => {
             const handleDeleteContent = async (content_id: number) => {
-              const results = apiCalls
```

Comment:
I see it as more modularised! There are few important principles of good design at play:
* lower level components should be like pure functions. Fully deterministic by arguments. It makes it easier to test and reuse code.
* all dependencies should be at the same level of abstraction. We had some calls to api.tsx from `page.tsx` and some from `/components`. This makes it hard to reason changes.
* Coupling should be at a high level. All low level components should just use code within the module and should not know the inner workings of other module. Allows for easier refactors. 

Happy to talk more. There is a good chapter Pragmatic Programmer and Clean Code that I can try to dig up.

In the interest of time, I'd rather leave it here. You'll be reviewing it anyway and it's all going to get squashed as one change to main anyway in the end.

Reply:
Makes sense, thanks for the explanation. That would've been easy for me to see on Python but it felt different here because it felt more unfamiliar. Thanks for picking this up!